File: my-folder/0776-n-ary-tree-postorder-traversal/solution.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public List<Integer> postorder(Node root) {
        List<Integer> res = new ArrayList<>();
        if (root == null) {
            return res;
        }
        bfs(root, res);
        return res;
    }

    public void bfs(Node root, List<Integer> list) {
        if (root == null) {
            return;
        }
        List<Node> children = root.children;
        for (Node child : children) {
            bfs(child, list);
        }
        list.add(root.val);
    }
}

File: my-folder/0776-n-ary-tree-postorder-traversal/solution.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        self.res = []
        def bfs(node):
            if not node: return

            children = node.children

            for child in children:
                bfs(child)

            self.res.append(node.val)

        if not root: return self.res

        bfs(root)

        return self.res
            

